.github/workflows/static-binary.yml

```diff
@@ -0,0 +1,85 @@
+name: "Static Binary"
+on:
+  repository_dispatch:
+    types: static-binary
+  push:
+    branches:
+      - master
+  pull_request:
+    paths:
+      - '.github/workflows/static-binary.yml'
+
+jobs:
+  manual:
+    if: github.event_name == 'repository_dispatch' || github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    name: Manual Static Binary (${{ github.event.client_payload.ref == '' && 'master' || github.event.client_payload.ref }}/${{ github.event.client_payload.args == '' && 'vast-pro' || github.event.client_payload.args }})
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v1
+      with:
+        ref: ${{ github.event.client_payload.ref == '' && 'master' || github.event.client_payload.ref }}
+
+    - name: Install Nix
+      uses: cachix/install-nix-action@v10
+
+    - name: Setup Cachix
+      uses: cachix/cachix-action@v6
+      with:
+        name: vast
+        signingKey: '${{ secrets.CACHIX_VAST_SIGNING_KEY }}'
+
+    - name: Build a Static Binary
+      run: |
+        STATIC_BINARY_TARGET=${{ github.event.client_payload.args }} nix/static-binary.sh --use-head
+
+    - name: Create Paths
+      id: create_paths
+      run: |
+        BUILD_DIR="build"
+        echo "::set-output name=build_dir::${BUILD_DIR}"
+        ARTIFACT_NAME=$(ls "$BUILD_DIR" | grep "vast-pro.*.tar.gz")
+        echo "::set-output name=artifact_name::${ARTIFACT_NAME}"
+
+    - name: Upload Artifact to Github
+      uses: actions/upload-artifact@v1
+      with:
+        name: "${{ steps.create_paths.outputs.artifact_name }}"
+        path: "${{ steps.create_paths.outputs.build_dir }}/${{ steps.create_paths.outputs.artifact_name }}"
+
+  push_to_master:
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    name: Master Static Binary (${{ matrix.args }})
```

Comment:
I don't think `${{ matrix.args }}` resolves to something that looks good in the left column of the actions tab, right?